test/fuzz/lua/test_mvcc.lua

```diff
@@ -0,0 +1,872 @@
+--[[
+The test for Tarantool allows you to generate random interleaved (concurrent)
+transactions for the memtx and vinyl engines, and verify the consistency of
+their execution w.r.t. serializability.
+
+All random operations and settings depend on the seed, which is set in the
+`seed` global test parameter.
+
+The test executes infinitely in a round fashion. Each round a number of
+transactions are executed, and the results of their execution are saved. Then,
+the successfully committed transactions are re-executed serially, based on the
+order in which they were committed, and the results of the serial execution are
+compared with the results of the interleaved (concurrent) execution. If a
+discrepancy is found, the test fails and the current execution round is dumped,
+as described below. Similar oracle-based approaches have been proposed by
+Zu-Ming Jiang et al. [1] and Wensheng Dou et al. [2].
+
+The test uses luatest's working directory, defined by the environment variable
+`VARDIR` (by default, it is `/tmt/t`). In case a discrepancy is found, the test
+generates two files, `repro.lua` and `serial.lua`, to facilitate the debugging
+process. The former file contains the transactions that were executed in the
+current round. The latter file contains the serial transaction execution
+schedule for the current round.
+
+The parameters for the test can be tweaked below.
+
+Usage: tarantool test_mvcc.lua
+
+[1]: https://www.usenix.org/conference/osdi23/presentation/jiang
+[2]: https://dl.acm.org/doi/10.1109/ICSE48619.2023.00101
+]]
+
+local fio = require('fio')
+local json = require('json')
+local log = require('log')
+local has_luatest = pcall(require, 'luatest')
+if not has_luatest then
+    print('luatest library is required')
+    os.exit(1)
+end
+local server = require('luatest.server')
```

Comment:
`luatest.server` is a `popen` with syntax sugar. Don't you want to replace it with popen and make test dependency free?

Reply:
okay